Semantic definitions for alignment (business terminology) (#130)
bSI-InfraRoom/IFC-Specification: IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/Entities/IfcAlignment/Documentation.md

> For the purposes of IFC the English term "alignment" defines three essentially separate but closely interconnected concepts.

Author: Thanks for this feedback.

**Reference system for linear positioning**

An alignment is used to define a reference system to position elements mainly for linear construction works, such as roads, rails, bridges, and others. The relative positioning along the alignment is defined by the linear referencing methodology.  
  
> NOTE&nbsp; See ISO 19148 Geographic information &ndash; Linear referencing for general definitions about linear referencing.  

**Kinematic perspective**

In the kinematic perspective focus is on the safe and optimized movement of a vehicle under the constraints induced by changes in the direction of the horizontal and the vertical layout.

**Geometric perspective**

In the geometric perspective the focus is on the proper placement of horizontal and vertical segments to connect certain points along a proposed path. A huge body of knowledge has been developed over a long period of time, in many aspects predating the availability of modern computers and their software.

**State of the art in contemporary engineering**

1. Contemporary engineering usually establishes first a horizontal layout in a properly projected plane.

> 2. In a second step the vertical profile (i.e. sequence of segments with constant gradients) is added.

Reviewer: Is it only a sequence of constant gradients? When are the ones with changing gradients added?

Author: Well spotted. Will modify.

3. In the rail domain in most cases a cant layout is added to the horizontal layout to compensate a part of the unwanted lateral acceleration.

> 4. In a final step the proposed layout is checked against a defined set rules, formulas and threshold to guarantee the conformance against the regulation.

Reviewer: ```suggestion
4. In a final step the proposed layout is checked against a defined set of rules, formulas and thresholds to guarantee the conformance against the regulation.
```

> The sequence of the steps might change from case to case and might be repeated one or more times to achieve the economic objectives and fullfill regulatory safety requirements. 

Reviewer: ```suggestion
The sequence of steps might change from case to case and might be repeated one or more times to achieve the economic objectives and fulfill regulatory safety requirements. 
```

> Contemporary alignment design itself implements almost always a 2.5 dimension approach. 

Reviewer: ```suggestion
Contemporary alignment design almost always implements a 2.5 dimension approach. 
```

> The resulting and documented geometry might be very precise or just good enough to meet safety thresholds. This depends on factors like priorities of the management, date of the design - existing alignments might have been more then 50 years ago - or software tools used. Working with legacy data in a high precision BIM model requires a good understanding of these factors.

Reviewer: ```suggestion
The resulting and documented geometry might be very precise or just good enough to meet safety thresholds. This depends on factors like priorities of the management, date of the design - existing alignments might have been designed more then 50 years ago - or software tools used. Working with legacy data in a high precision BIM model requires a good understanding of these factors.
```

**Distinction between business modeling and IFC core geometry**

According to IFC modeling principles alignment entities are organised in two large parts. The two parts work together, but they can also be used independently from each other

1. Business aspects of alignment

> 2. Representation with IFC geometry

Reviewer: ```suggestion
2. Representation with the IFC geometry resources
```

> **Business aspects of alignment:** Here the focus is a on a schema structure as close to business terminology as possible. Here it is possible to have a very detailed segment structure with many domain specific properties attached. Examples for domain specific properties are design speed or cant deficiency.

Reviewer: ```suggestion
**Business aspects of alignment:** Here the focus is a on a schema structure as close to business terminology as possible. It is possible to have a very detailed segment structure with many domain specific properties attached. Examples for domain specific properties are design speed or cant deficiency.
```

> **Representation with IFC geometry:** Here the focus is on using as much of the established IFC geometry entities as possible. A mapping between Business aspects and IFC geometry is proposed.

Reviewer: ```suggestion
**Representation with IFC geometry resources:** Here the focus is on using as much of the established IFC geometry entities as possible. A mapping between Business aspects and IFC geometry is proposed.
```

**IFC modelling**
  
In IFC a single alignment must have:  
  

> \* a horizontal alignment defined in the x/y plane of the engineering coordinate system  

Reviewer: ```suggestion
* a horizontal alignment defined in the x/y plane of the engineering coordinate system  
```

A single alignment may have: 

> \* an accompanying vertical alignment, defined along the horizontal alignment in the distance along / z coordinate space  

Reviewer: ```suggestion
* an accompanying vertical alignment, defined along the horizontal alignment in the distance along / z coordinate space  
```

> \* a relative alignment, defined as a span within another alignment and/or at constant or variable offsets  

Reviewer: ```suggestion
* a relative alignment, defined as a span within another alignment and/or at constant or variable offsets  
```

> \* a 3D alignment, either computed from the horizontal and vertical alignment, or extracted from geospatial data.  

Reviewer: ```suggestion
* a 3D alignment, either computed from the horizontal and vertical alignment, or extracted from geospatial data.  
```

1. Business aspects of alignment
2. Representation with the IFC geometry resources

**Business aspects of alignment:** Here the focus is a on a schema structure as close to business terminology as possible. It is possible to have a very detailed segment structure with many domain specific properties attached. Examples for domain specific properties are design speed or cant deficiency.

**Representation with IFC geometry resources:** Here the focus is on using as much of the established IFC geometry entities as possible. A mapping between Business aspects and IFC geometry is proposed.

**IFC modelling**
  
In IFC a single alignment must have:  
  
* a horizontal alignment defined in the x/y plane of the engineering coordinate system  

A single alignment may have: 

* an accompanying vertical alignment, defined along the horizontal alignment in the distance along / z coordinate space  
* a relative alignment, defined as a span within another alignment and/or at constant or variable offsets  
* a 3D alignment, either computed from the horizontal and vertical alignment, or extracted from geospatial data.  
  

> Alignments may be aggregated into referents (_IfcReferent_) or derivative alignments. Derivative alignments may be used to indicate dependent alignments, such as an alignment for a bridge that is relative to a parent alignment for a road, where the child _IfcAlignment_ may have _Axis_ set to _IfcOffsetCurveByDistances_ that starts and ends at a span within the extent of the original _IfcAlignment_ defined at the _Axis_ of the parent _IfcAlignment_.  

Reviewer 2: What does _Axis_ refer to here?

Author: This is previous text. Please provide a better description based on the final schema. @larswik @czapplitec ?

Reviewer: New issue already created from one of the comments below.

  
Alignments may be assigned to groups using _IfcRelAssignsToGroup_, where _IfcGroup_ or subtypes may capture information common to multiple alignments.  
  

> Supported representations of <span class="self-ref">IfcAlignment</span>.Axis are:  

Reviewer 2: _Axis_ again. Should this instead be "Supported shape representations for IfcAlignment are (RepresentationIdentifier='Axis" and RepresentationType="Curve2D/Curve3D"):
...

Author: This is previous text. Please provide a better description based on the final schema. @larswik @czapplitec ?

Reviewer: We can address this after merging this PR. It should be a smaller change.

  
* _IfcGradientCurve_ as a 3D horizontal and vertical alignment (represented by their alignment segments)  
* _IfcCompositeCurve_ as a 2D horizontal alignment (represented by its horizontal alignment segments) without a vertical alignment  
* _IfcOffsetCurveByDistances_ as a 2D or 3D curve defined relative to an _IfcAlignmentCurve_ or another _IfcOffsetCurveByDistances_  
* _IfcSegmentedReferenceCurve_ as a 3D curve defined relative to an _IfcGradientCurve_ to incorporate the application of cant 
* _IfcPolyline_ as a 3D alignment by a 3D polyline representation (such as coming from a survey)  
* _IfcPolyline_ as a 2D horizontal alignment by a 2D polyline representation (such as in very early planning phases or as a map representation)  
  

> > NOTE&nbsp; Although _Axis_ is an _IfcCurve_ base type, only derived types _IfcGradientCurve_, _IfcSegmentedReferenceCurve_, _IfcOffsetCurveByDistances_, and _IfcPolyline_ are meant to be supported types. Derivative specifications (Model View Definitions) may expand this set to include additional curve types.

Reviewer 2: _Axis_

Author: This is previous text. Please provide a better description based on the final schema. @larswik @czapplitec ?
